packages/material-ui/src/Chip/Chip.js

```diff
@@ -378,12 +504,11 @@ const Chip = React.forwardRef(function Chip(props, ref) {
 
   let icon = null;
   if (iconProp && React.isValidElement(iconProp)) {
-    icon = React.cloneElement(iconProp, {
-      className: clsx(classes.icon, iconProp.props.className, {
-        [classes.iconSmall]: small,
-        [classes[`iconColor${capitalize(color)}`]]: color !== 'default',
-      }),
-    });
+    icon = (
```

Comment:
Watch out, here you are creating a wrapper around the original component. We have two options of how to solve this:
1. Add the styles in the root using the class selector for this slot - we would be increasing the specificity with this :\
2. Another option would be to use the `as` prop to render the `iconProp`, something like:
```
<ChipIcon 
  {...iconProp.props} 
  as={iconProp.type} 
  styleProps={styleProps}
  className={clsx(classes.icon, iconProp.props.className)}
/> 
```
Here is an example with working example - https://codesandbox.io/s/sad-wood-m4vfh?file=/src/App.js


I would vote for option 2, it should be the one without introducing any breaking changes. cc @oliviertassinari let me know if you agree.

Reply:
This case with the icon looks identical to the case with the avatar. I think that we should increase the specificity anytime we use cloneElement. I can search the issue that we had it the past with the avatar that lead to increasing the specificity.

Reply:
https://github.com/mui-org/material-ui/pull/18156

Reply:
We can always try option 2 and see how it goes

Reply:
I am always more scared to introduce bigger specificity, I'd say let's try not to if we see the problem we can do it later.. I am pretty sure we are soon going to receive complain if something is not working.. :)

Reply:
Ah I see what you are saying, if we don't increase the specificity it will be the same as the original component, and that creates problems, am I right?

Reply:
If that's the case then let's go with the bigger specificity

Reply:
@natac13 sorry for the confusion let's go with increasing the specificity, the root would contain the styles for the `avatar`, `icon`, and `delteIcon`. The `label` is anyway a different component.

Reply:
No worries. Ill make the changes

Reply:
Basically, if we want to support custom icons, we need to consider the following:

- With emotion, we get the specificity order with prop cascading: flattening should work
- With JSS, and sc with are getting the specificity with import order, https://github.com/mui-org/material-ui/issues/16374#issuecomment-546139848 shares an issue we had in the past. It was related to dead-code elimination. In order to get it to work, developers have to import the icon and the chip in the right order, it's error-prone and not obvious. @dtassone experienced this problem first hand with the data grid recently.
- With CSS we are getting injection order, developers will configure it to have more specificity, they won't have any luck.

Reply:
@oliviertassinari would be nice if you can also do a final review - @natac13 already implemented the changes with increasing the specificity, looks good to me